realm/realm-library/src/androidTestObjectServer/java/io/realm/UserTests.java

```diff
@@ -68,6 +76,27 @@ public void currentUser_returnsNullIfUserExpired() {
         assertNull(User.currentUser());
     }
 
+    // `all()` returns an empty list if no users are logged in
+    @Test
+    public void all_empty() {
+        Collection<User> users = User.all();
+        assertTrue(users.isEmpty());
+    }
+
+    // `all()` returns only valid users. Invalid users are filtered.
+    @Test
+    public void all_validUsers() {
+        // Add 1 expired user and 1 valid user to the user store
+        UserStore userStore = new SharedPrefsUserStore(InstrumentationRegistry.getContext());
```

Comment:
The SharedPreference is a Android concept which require the Context. The `UserStore` is our general interface that enable us to swap implementations for the relevant platforms. The `UserStore` do not have any knowledege about Context